network/gossip/libp2p/test/testUtil.go

```diff
@@ -0,0 +1,216 @@
+package test
+
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/phayes/freeport"
+	"github.com/rs/zerolog"
+	testifymock "github.com/stretchr/testify/mock"
+
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/module/mock"
+	"github.com/onflow/flow-go/network/codec/json"
+	"github.com/onflow/flow-go/network/gossip/libp2p"
+	"github.com/onflow/flow-go/network/gossip/libp2p/topology"
+	protocol "github.com/onflow/flow-go/state/protocol/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+var rootBlockID = unittest.IdentifierFixture().String()
+
+// generateIDs generate flow Identities with a valid port and networking key
+func generateIDs(n int) ([]*flow.Identity, []crypto.PrivateKey, error) {
```

Comment:
I would suggest passing a `t *testing.T` here as a parameter. Then instead of returning the error, you can fail the test at the line it happens. It would be easier to track.

Reply:
makes sense..will change it to accept t and not return an error